src/hypertable.c

```diff
@@ -337,10 +337,32 @@ chunk_store_entry_free(void *cse)
 	MemoryContextDelete(((ChunkStoreEntry *) cse)->mcxt);
 }
 
+static bool
+table_is_compressed_or_materialization(Hypertable *ht)
+{
+	ContinuousAggHypertableStatus status = ts_continuous_agg_hypertable_status(ht->fd.id);
+	if (ht->fd.compressed || status == HypertableIsMaterialization)
+		return true;
+	else
+		return false;
+}
+
+static ScanFilterResult
+filter_exclude_compressed_and_materialization(TupleInfo *ti, void *data)
+{
+	Hypertable *ht = ts_hypertable_from_tupleinfo(ti);
+
+	if (table_is_compressed_or_materialization(ht))
+		return SCAN_EXCLUDE;
+	else
```

Comment:
Just nit, I would suggest just `return SCAN_INCLUDE` at the end of function and omit this else statement. Sometimes different compiler versions warns about missing return at the end of a function